### ISSUE TITLE
SUPERLU_DIST Wrapper

### ISSUE THREAD

**Issue by ChrisRackauckas** [SciML/LinearSolve.jl#49]

**Comment by ChrisRackauckas** [comment]
@Wimmerer I noticed you were working on some?

**Comment by Wimmerer** [comment]
Yep almost done, just packaging working stuff from local machine.

**Comment by Wimmerer** [comment]
Also finishing up Magma which Yonatan worked on

**Comment by AJ0070** [comment]
@ChrisRackauckas I have the `LinearSolveSuperLUDIST` wrapper working for the real-valued sparse path, and the new `test/superludist` group passes locally once `SuperLUDIST.jl` is used from a compatible source checkout.

The remaining blocker is dependency resolution against released packages:

- `LinearSolve` requires `DocStringExtensions = 0.9.x`
- released `SuperLUDIST 0.3.x` still restricts `DocStringExtensions` to `0.8.x`

So a clean `Pkg.instantiate()` for `test/superludist` fails before the test even starts when using the registry version of `SuperLUDIST`.

I wanted to ask what you would prefer here:

1. Wait for a `SuperLUDIST.jl` release that includes the compat fix, and leave the `LinearSolveSuperLUDIST` test group pending until then.
2. Temporarily point `test/superludist/Project.toml` at a specific `SuperLUDIST` branch/commit so CI can run now.
3. Do something else on the `LinearSolve` side.

There is one other upstream issue: the `ComplexF64` replicated `SuperLUDIST` path currently segfaults in `SuperLUDIST.jl` finalization, so on the `LinearSolve` side I changed the wrapper to reject complex inputs cleanly and only support real-valued inputs for now.

**Comment by ChrisRackauckas** [comment]
Make a PR that updates the compats and I'll review there.
